builder/azure/arm/builder.go

```diff
@@ -82,11 +83,21 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 		return nil, err
 	}
 
+	resolver := newResourceResolver(azureClient)
+	if err := resolver.Resolve(b.config); err != nil {
+		return nil, err
+	}
+
 	b.config.storageAccountBlobEndpoint, err = b.getBlobEndpoint(azureClient, b.config.ResourceGroupName, b.config.StorageAccount)
 	if err != nil {
 		return nil, err
 	}
 
+	endpointConnectType := PublicEndpoint
```

Comment:
The intent was always to use the private IP.  If the VMs are on the same subnet there should be no need to use a public IP, and there may not even be a public IP.

The [original request][1] for the feature was to support private IP access only.

[1]: https://github.com/Azure/packer-azure/issues/193